featurebyte/service/preview.py

```diff
@@ -436,6 +437,68 @@ async def preview_featurelist(
 
         return dataframe_to_json(result)
 
+    async def preview_target(
+        self, target_preview: TargetPreview, get_credential: Any
+    ) -> dict[str, Any]:
+        """
+        Preview a Target
+
+        Parameters
+        ----------
+        target_preview: TargetPreview
+            TargetPreview object
+        get_credential: Any
+            Get credential handler function
+
+        Returns
+        -------
+        dict[str, Any]
+            Dataframe converted to json string
+        """
+        graph = target_preview.graph
+        target_node = graph.get_node_by_name(target_preview.node_name)
+        operation_struction = target_preview.graph.extract_operation_structure(target_node)
+
+        # We only need to ensure that the point in time column is provided,
+        # if the feature aggregation is time based.
+        (
+            point_in_time_and_serving_name_list,
+            updated,
+        ) = PreviewService._update_point_in_time_if_needed(
+            target_preview.point_in_time_and_serving_name_list, operation_struction.is_time_based
+        )
+
+        request_column_names = set(point_in_time_and_serving_name_list[0].keys())
+        feature_store, session = await self._get_feature_store_session(
+            graph=graph,
+            node_name=target_preview.node_name,
+            feature_store_name=target_preview.feature_store_name,
+            get_credential=get_credential,
+        )
+        parent_serving_preparation = (
+            await self.entity_validation_service.validate_entities_or_prepare_for_parent_serving(
+                graph=graph,
+                nodes=[target_node],
+                request_column_names=request_column_names,
+                feature_store=feature_store,
+            )
+        )
+        preview_sql = get_feature_or_target_preview_sql(
+            request_table_name=f"{REQUEST_TABLE_NAME}_{session.generate_session_unique_id()}",
+            graph=graph,
+            nodes=[target_node],
+            point_in_time_and_serving_name_list=point_in_time_and_serving_name_list,
+            source_type=feature_store.type,
+            parent_serving_preparation=parent_serving_preparation,
+            skip_tile_generation=True,
+        )
+        result = await session.execute_query(preview_sql)
+        if result is None:
+            return {}
+        if updated:
+            result = result.drop(SpecialColumnName.POINT_IN_TIME, axis="columns")
+        return dataframe_to_json(result)
```

Comment:
Curious why don't we reuse the feature preview route? 🤔

Reply:
ah possible now, i previously had a separate helper function that was being called from here, but now that that's cleaned up, i should be able to reuse the feature preview route here too i tihnk.